### DISCUSSION TITLE
Bandwidth Part (BWP) Support

### DISCUSSION THREAD

**Discussion by ryanwwest** [srsran/srsRAN_Project#52] [General Help] [answered]
Hi there, I'm curious as to how well [NR BWPs](https://www.sharetechnote.com/html/5G/5G_CarrrierBandwidthPart.html) are currently supported in the gNB? Specifically I'm wondering if you can define (in configuration files or dynamically) multiple BWPs (e.g., Initial, Dedicated, First Active, Regular types) and if so, how you can control which BWPs a particular connected UE should use? OAI has limited support for these ([see here](https://gitlab.eurecom.fr/oai/openairinterface5g/-/blob/develop/doc/RUN_NR_multiple_BWPs.md)) in config files, but doesn't yet have a stable way to dynamically switch between BWPs (you instead have to restart the gNB changing its static configuration). 

I don't see any documentation or discussions regarding BWPs for srsRAN but do see it mentioned a lot in code, so I wanted to check to see if there are any gotchas I should be aware of. Can it be configured at all yet? If support is very minimal, it would at least be good to know how I could change the amount of usable bandwidth for all UEs connected to a gNB, regardless of NR band, on 30kHz SCS (e.g. change from 40 MHz to 10 MHz to 100MHz).

I see [this config line](https://github.com/srsran/srsRAN_Project/blob/fbe73a49cd2af296e5f23597c928da9e4128b083/configs/gnb_rf_b200_tdd_n78_20mhz.yml#L18), channel_bandwidth_MHz, is likely related. Does this ultimately set the initial (and possibly only) BWP to the correct locationAndBandwidth value, and if so, how does it determine the starting PRB (or relative start/end position within the selected band)? Thanks.

**Comment by andrepuschmann** [comment, marked as answer]
We currently only support a single BWP as you already figured. This default BWP has the size of the `channel_bandwidth_MHz`. Supporting multiple BWP is on the roadmap but no estimated date of arrival.

**Reply by ItsGarand** [reply]
> We currently only support a single BWP as you already figured. This default BWP has the size of the `channel_bandwidth_MHz`. Supporting multiple BWP is on the roadmap but no estimated date of arrival.

Since it has been over two years since this question has been asked, has there been any movement on this feature?
